### PR TITLE
Extend binding

### PR DESCRIPTION
**Pull request by KuraiAndras**
- Update unity and dependencies
- Clean up warnings
- Make project buildable
- Make default backend IL2CPP with medium stripping
- Add `GetOrAddComponent` to components
- Add `CheckSerializedField` expression based overload
- `CheckSerializedField` only runs inside editor, also aggressively inlined
- Rename binding methods to `Bind`
- Make bindings use expressions to resolve the property name
- Add bindings to `UnityEvent`
- Bind to `INotifyCollectionChanged`
- Add generic `ListView` script

### CODE REVIEW

File: Source/AutSoft.UnitySupplements/Assets/AutSoft.UnitySupplements.Vitamins/AutSoft.UnitySupplements.Vitamins.asmdef [hunk 1]
```diff
@@ -1,7 +1,9 @@
 {
     "name": "AutSoft.UnitySupplements.Vitamins",
     "rootNamespace": "",
-    "references": [],
+    "references": [
+        "Injecter.Unity"
```
Comment (Vikesz12) [inline Source/AutSoft.UnitySupplements/Assets/AutSoft.UnitySupplements.Vitamins/AutSoft.UnitySupplements.Vitamins.asmdef:5]: Ezt a doksiba is bele kell írni akkor gondolom

File: Source/AutSoft.UnitySupplements/Assets/AutSoft.UnitySupplements.Samples/VitaminSamples/BindingSamples/EditableListItem.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                x => x.Title,
+                t => _titleText.text = t,
+                i => i
+            );
+
+            Data.Bind
+            (
+                gameObject,
+                _numberText.onValueChanged,
+                x => x.Number,
```
Comment (Vikesz12) [inline Source/AutSoft.UnitySupplements/Assets/AutSoft.UnitySupplements.Samples/VitaminSamples/BindingSamples/EditableListItem.cs:34]: Valahogy ez a sorrend nem tűnik intuitívnak

**Review by Vikesz12 — COMMENTED** [review]
image

Update package verzió?

**Comment by KuraiAndras** [comment]
ListView.Initialize should take a lambda to init

**Pull request closed without merging**